pkgs/top-level/all-packages.nix

```diff
@@ -19486,6 +19486,7 @@ in
   termtosvg = callPackage ../tools/misc/termtosvg { };
 
   tesseract = callPackage ../applications/graphics/tesseract { };
+  tesseractWithoutData = callPackage ../applications/graphics/tesseract { enableLanguages = false; };
```

Comment:
1. Maybe just use `[]` for the no-languages case?

2. It is already available as both `tesseract.tesseractWithoutData` and (if you agree with point 1) `tesseract.override {enableLanguages = [];}`, does it need a top-level attribute or maybe just a top-level _comment_?

```
# you can pass `enableLanguages` to `tesseract` to install tessearct with fewer supported languages and save space,
# or even pass `enableLanguages = []` and then supply the language packs via environment.
```

Reply:
1. I have no preference, I'll let you decide.
2. We need a top-level attr so that the no-languages drv is cached. Without it, we would incur a full compilation when `enableLanguages` is customized.

Reply:
I think `[]` reduces the number of types (list or null seems to be a reasonable union type, list of null or false boolean should provide some value). So I would slightly prefer `[]`

Given the size of the compiled part (less than a single language model), maybe symlink the files from there and only copy the files that need to be edited to replace the path? I think `sed` can even break symlinks automatically in the inplace mode. Then the binary build becomes a dependency of the toplevel attribute.

(I am not sure if Hydra actually keeps the build depepndencies — then the entire point is moot — but even if not, not a large problem)

Reply:
I was just typing up exactly what you proposed here. :smile_cat: 
With this option, the build script of `tesseractWithData` will be a bit more complex, but we lose the slightly unelegant extra top-level attr. I think that's preferable.
I'll have breakfast and then implement it. Thanks for your very thoughtful feedback!

Reply:
[Done.](https://github.com/NixOS/nixpkgs/pull/52379/commits/b02785bfc4b61b1f1e4b09fabba49d0cd1a77fba)